introduction.Rmd

```diff
@@ -184,6 +184,309 @@ We call the combination of these two steps the __quote and unquote__ pattern. Th
 Most of the time though, we don't need to create blueprints manually. We'll get them by quoting the arguments supplied by users. This gives your functions the same usage and feel as tidyverse verbs.
 
 
+# Do you need tidy eval? {#sec:do-you-need}
+
+In the field of computer science, frameworks like tidy evaluation are known as metaprogramming. Modifying the blueprints of computations amounts to programming the program, i.e. metaprogramming. In other languages, this type of approach is often seen as a last resort because it requires new skills and might make your code harder to read. Things are different in R because of the importance of data masking functions, but it is still good advice to consider other options before turning to tidy evaluation. In this section, we review several strategies for solving programming problems with tidyverse packages.
+
+Before diving into tidy eval, make sure to know about the fundamentals of programming with the tidyverse:
+
+* [Fixed column names](#sec:fixed-colnames). A solid function taking data frames with fixed column names is better than a brittle function that uses tidy eval.
```

Comment:
Verb-fiy this bullet, or noun-ify the next one for list agreement. e.g. "Using fixed column names" or "Automatic loops"

Reply:
Add a bullet for "Scoped dplyr variants" up here?

Reply:
I think I prefer to keep them as is because they are section titles. I plan to transform the next bullet list to section titles as well once they're written. Does that make sense?

> Add a bullet for "Scoped dplyr variants" up here?

It's part of automating loops columnwise.

Reply:
Makes sense. Maybe mention scoped verbs in the "automating loops" bullet?

Reply:
I would like to avoid mentioning the scoped verbs too much because they are likely to be retired in the next version. We're thinking to replace them all with a single (ad)verb: https://github.com/tidyverse/dplyr/issues/4169. That's why the bullet mentions "columnwise mapping", which is the underlying concept that is here to stay :)

Reply:
Gotcha!